pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -98,6 +98,9 @@ type InstallerController struct {
 	clock            clock.Clock
 	installerBackOff func(count int) time.Duration
 	fallbackBackOff  func(count int) time.Duration
+
+	// shouldRevisionInstall is a callback function that determines whether a new revision should be installed
+	shouldRevisionInstall func() (bool, error)
```

Comment:
I think we should follow the same precondition pattern that we have in the static resource controller: https://github.com/openshift/library-go/blob/master/pkg/operator/staticresourcecontroller/static_resource_controller.go#L55-L59. Meaning that we would allow for any condition to be defined to gate the revision process.

Also, we should pass on a context in that method and propagate it all the way down in case we need to cancel it for some reasons. Especially in https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/ceohelpers/bootstrap.go#L139 and maybe some other functions that don't take it yet.

Reply:
good idea, @jubittajohn can we create it slightly different like this?

> type StaticPodInstallerPreconditionsFuncType func(ctx context.Context) (bool, error)

and then the struct value could look like this

>	installPrecondition StaticPodInstallerPreconditionsFuncType

that way is more consistent with the other packages.